index.src.html

```diff
@@ -830,18 +856,29 @@ <h4 id="algorithm-create" algorithm>Create a `Credential`</h4>
             types in order to support a "sign-up" use case. For the moment, though, we're punting
             on that by restricting the dictionary to a single entry.
 
-    5.  Let |p| be [=a new promise=].
+    6.  Let |p| be [=a new promise=].
 
-    6.  Run the following steps [=in parallel=]:
+    7.  Let |origin| be |settings|'s [=environment settings object/origin=].
 
-        1.  Let |r| be the result of executing |interfaces|[0] {{Credential/[[Create]](options)}}
-            internal method on |options|.
+    8.  Run the following substeps [=in parallel=]:
 
-        2.  If |r| is an [=exception=], [=reject=] |p| with |r|.
+        1.  Let |r| be the result of executing |interfaces|[0]'s {{Credential/[[Create]](origin, options)}}
+            internal method given |origin| and |options|.
 
-            Otherwise, [=resolve=] |p| with |r|.
+        2.  If |r| is an [=exception=], [=reject=] |p| with |r|, and terminate these substeps.
 
-    7.  Return |p|.
+        3.  If |r| is a {{Credential}}, [=resolve=] |p| with |r|, and terminate these substeps.
+
+        4.  Assert: |r| is a <code>[=constructCredential=]</code> [=struct=].
```

Comment:
I think you've lost the possibility that |r| is `null`.

Reply:
yeah, i think you are correct.  Rather than an `Assert:` we can make this a Note I suppose. Given that `[[Create]]` returns various different things depending upon the situation, I prefer explicitly documenting for the reader what is occurring at this point.

Reply:
I think `Assert:` is correct; you just have to handle |r|==`null` in a previous step.

Reply:
ah.  i misunderstood your comment.  o_0